src/clj/disaster/middleware/oauth2_jwt.clj

```diff
@@ -0,0 +1,197 @@
+                       ;;invalid:
+                       "xnmcbvjksdhfwiuerfsdjbsdkjfghwileugkhjbvnxdvjbvwiuerhslkdjbvvklwl4iuhjxcvxnmbvkwerjlfhiwuerhsjdkdfkjbvwe4riefslkv-dlsfkjhwpoiefhcvsdjkhvowpefwoeifhv_sdøflkhjwpeoifhsvøkl"})))
+;; d is the private key component of a json web key
+(s/def ::d (s/with-gen string?
+             #(s/gen #{;; Valid:
+                       "PJrXSYLiYRebbJN4yHujP3LfoHzCEnVh3Jl2FN9KaWK260HmROQYZG-sPQ5Bwqg-bz1xbyE1dQfSsuBy-3LqHrqM-ilsvcNZqQEY9R52d9D6kXmTSNMHx-3jGQ0SeO0eIFMHffLHOomvECPEKZkSPB65rijLcKQKmbnA_OlF_EE"
+                       ;; Invalid:
+                       "xnmcbvjksdhfwiuerfsdjbsdkjfghwileugkhjbvnxdvjbvwiuerhslkdjbvvklwl4iuhjxcvxnmbvkwerjlfhiwuerhsjdkdfkjbvwe4riefslkv-dlsfkjhwpoiefhcvsdjkhvowpefwoeifhv_sdøflkhjwpeoifhsvøkl"})))
+(s/def ::e string?)
+(s/def ::claims (s/keys :opt-un [::exp
+                                 ::scope
+                                 ::scopes
+                                 ::sub]))
+(s/def ::jwt (s/nilable (s/with-gen (s/and string?
+                                           #(re-matches jwtregex %))
+                          #(s/gen #{(jwt/sign (gen/generate (s/gen ::claims))
+                                              "secret")}))))
+(s/def ::jwt-header (s/keys :req-un [::kid ::kty]))
+(s/def ::jwk (s/keys :req-un [::kty ::e ::n ::kid]))
+(s/def ::public-key keys/public-key?)
+(s/def ::private-key keys/private-key?)
+(s/def ::key (s/keys :req-un [::public-key]
+                     :opt-un [::private-key]))
+(s/def ::key-store (s/map-of ::kid
+                             ::key))
+(s/def ::resource (s/with-gen #(or (instance? java.net.URL %)
+                                   (instance? (class (char-array "")) %))
+                    ;; Always use local resources to avoid spamming actual servers
+                    #(s/gen #{(resource "jwks.json")
+                              (resource "jwks-other.json")})))
+(s/def ::jwks-url (s/with-gen (s/or :url  :invetica.uri/absolute-uri-str
+                                    :resource ::resource)
+                    ;; Always use local resources to avoid spamming actual servers
+                    #(s/gen #{(resource "jwks.json")
+                              (resource "jwks-other.json")})))
+
+(s/fdef jwks-edn->keys
+  :args (s/cat :jwks (s/coll-of ::jwk :type vector?))
+  :ret ::key-store)
+
+(defn- jwks-edn->keys
+  "Transform a vector of json web keys into a map of kid -> key pairs where each key is a map
+  of :public-key and optionally :private-keys."
+  [json-web-keys]
+  (->> json-web-keys
+       :keys
+       (filter #(= (:kty %) "RSA"))
+       (group-by :kid)
+       (fmap first)
+       (fmap #(assoc {}
+                     :public-key (buddy-jwk/jwk->public-key %)
+                     :private-key (buddy-jwk/jwk->private-key %)))))
```

Comment:
`fmap` got moved to a separate package so not sure where you're getting this from

Reply:
`[clojure.algo.generic.functor :refer [fmap]]`
I didn't do it \o/